### PR TITLE
Add support for outputting 256 colors and also add documentation about default command functionality

### PR DESCRIPTION
**Pull request by kodie**
As the title says, this PR allows us to use 256 colors in our custom styles like so:

```php
\Ahc\Cli\Output\Color::style('mystyle', [
  'fg' => '38;5;57',
  'bg' => '48;5;82'
]);
```

and all it took was removing typecasting on the `fg` style value (allowing us to pass strings) and doing a check on the `bg` style value to see if it's being passed an interval or not, if it is (default behavior) then add 10 to it and use it just the same before, but if it's not an interval (like a string) then just use that value, allowing us to use 256 colors.

This opens up the world of setting custom colors:

```php
\Ahc\Cli\Output\Color::style('indigo', ['fg' => '38;5;57']);
\Ahc\Cli\Output\Color::style('lime', ['fg' => '38;5;190']);
$writer = new Writer();
$writer->colors("<indigo>Loading</end> <lime>{$input_file}</end>...", true);
```

I added a small example of this to the readme and also added documentation about the default command functionality as that seemed to be missing.

### CODE REVIEW

**Comment by kodie** [comment]
If we wanted to get real fancy, we could add a `256` method to the `Color` class that adds the prefix for us that could work like this:

```php
\Ahc\Cli\Output\Color::style('mystyle', [
  'fg' => \Ahc\Cli\Output\Color::256(57),
  'bg' => \Ahc\Cli\Output\Color::256(82, true)
]);
```

the second parameter would set if it's a background or not (defaults to `false` meaning foreground).

Just an idea, thoughts?

**Comment by kodie** [comment]
Orrrrr, we could have it be it's own style property that can fallback for terminals that don't support 256:

```php
\Ahc\Cli\Output\Color::style('mystyle', [
  'bg' => Ahc\Cli\Output\Color::CYAN, // used if no 256-support
  'fg' => Ahc\Cli\Output\Color::WHITE, // used if no 256-support
  'bg_256' => 82,
  'fg_256' => 57
]);
```

Just another idea. But I'm totally fine with just how the PR is now.

**Comment by adhocore** [comment]
hello, thank you for this idea/ feature/ pull. and yes we can add the color consts Color::RED6, Color::GREEN6 etc for common/major colors and static func `Color::c($code)` for the rest.

**Comment by adhocore** [comment]
cc @dimtrovich more new exciting pulls might need your look 😁 😉

**Comment by kodie** [comment]
I figured the first suggestion was slightly better than the second as it only adds two methods rather than changing the flow of things (text, style, mod, etc).

So I've implemented this the fg256 and bg256 methods like so:

```php
\Ahc\Cli\Output\Color::style('mystyle', [
  'fg' => \Ahc\Cli\Output\Color::fg256(57),
  'bg' => \Ahc\Cli\Output\Color::bg256(82)
]);
```

File: src/Output/Color.php [hunk 1]
```diff
@@ -95,6 +95,22 @@ public function info(string $text, array $style = []): string
         return $this->line($text, ['fg' => static::BLUE] + $style);
     }
 
+    /**
+     * Returns the color code for a 256 background color.
+     */
+    public function bg256(int $code)
```
Comment (adhocore) [inline src/Output/Color.php:101]: missing return type `: string` at the end

Comment (kodie) [reply src/Output/Color.php]: Updated!

**Review by adhocore — DISMISSED** [review]
looks great, thanks

**Comment by adhocore** [comment]
is it possible to share a screenshot of 256 colors using this code, as well as from the other pulls? :D

**Comment by kodie** [comment]
> is it possible to share a screenshot of 256 colors using this code, as well as from the other pulls? :D

I have a branch on my fork where I have been merging all of my PRs together for testing (https://github.com/kodie/adhocore-php-cli/tree/kodie/development).

Given this code:

```php
$app = new Ahc\Cli\Application('My App', $version);
$writer = new Ahc\Cli\Output\Writer();
\Ahc\Cli\Output\Color::style('indigo', array('fg' => \Ahc\Cli\Output\Color::fg256(57)));
\Ahc\Cli\Output\Color::style('spring', array('fg' => \Ahc\Cli\Output\Color::RED, 'bg' => \Ahc\Cli\Output\Color::bg256(49)));
$help = "<indigo>Hello</end>\n<spring>World</end>";
$app->help($writer->colorizer()->colors($help));
$app->handle($_SERVER['argv']);
```

and running
```sh
./bin/my-app --help
```

I get this:
Screenshot 2024-11-25 220049

**Review by adhocore — APPROVED** [review]

**Comment by adhocore** [comment]
thanks, looks good, lets see if @dimtrovich would like to have a look as well

**Comment by dimtrovich** [comment]
> thanks, looks good, lets see if @dimtrovich would like to have a look as well

For the moment I'm still at home. As soon as I get to the office, I'll take a closer look.

**Comment by kodie** [comment]
Here's a better screenshot of the additional colors as well as the customizable help styles from #111 

Screenshot 2024-11-25 232723

**Comment by kodie** [comment]
Just playing around with colors. Added even and odd capabilities to #111 

Screenshot 2024-11-26 000031

**Comment by dimtrovich** [comment]
it's cool, thanks.

**Comment by adhocore** [comment]
@kodie pull needs a rebase

**Comment by kodie** [comment]
> @kodie pull needs a rebase

@adhocore Done!

**Comment by adhocore** [comment]
@kodie it still shows merge conflict:
```
This branch has conflicts that must be resolved
Use the web editor or the  to resolve conflicts.
Conflicting files
src/Output/Color.php
tests/Output/ColorTest.php
```

besides that i think good to go

**Comment by kodie** [comment]
> @kodie it still shows merge conflict:
> 
> ```
> This branch has conflicts that must be resolved
> Use the web editor or the  to resolve conflicts.
> Conflicting files
> src/Output/Color.php
> tests/Output/ColorTest.php
> ```
> 
> besides that i think good to go

@adhocore I just pushed another update that should fix that

**Pull request merged**